lib/rsvp/-internal.js

```diff
@@ -30,6 +34,25 @@ export function getThen(promise) {
   }
 }
 
+export const TRY_CATCH_ERROR = new ErrorObject();
+
+let tryCatchCallback;
+function tryCatcher() {
```

Comment:
the original approach was benchmarked, and at the time it was faster. With this change I am not sure that is still true, we may also be able to switch to regular try/catch now (or soon) thanks to turbofan in v8.

Originally I used: https://github.com/stefanpenner/promise_benchmarks and +  https://github.com/petkaantonov/bluebird/tree/master/benchmark to come to these conclusions. It may be worth resurrecting that for this change?